src/jit/inline.cpp

```diff
@@ -169,14 +164,135 @@ InlineContext::InlineContext()
     , inlObservation(InlineObservation::CALLEE_UNUSED_INITIAL)
 #ifdef DEBUG
     , inlCallee(nullptr)
+    , inlTreeID(0)
+    , inlSuccess(true)
 #endif
 {
     // Empty
 }
 
+//------------------------------------------------------------------------
+// newRoot: construct an InlineContext for the root method
+//
+// Arguments:
+//   compiler - compiler doing the inlining
+//
+// Return Value:
+//    InlineContext for use as the root context
+//
+// Notes:
+//    We leave inlCode as nullptr here (rather than the IL buffer
+//    address of the root method) to preserve existing behavior, which
+//    is to allow one recursive inline.
```

Comment:
This is a bit confusing to me - why wouldn't a recursive inline get a new InlineContext (i.e. one that is not a root context)?